### ISSUE TITLE
[bug] Different shape/size glitching/aliasing artifacts with icon button widgets when using custom icons

### ISSUE THREAD

**Issue by kennymc-c** [unfoldedcircle/bug-reports#418]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Description

I noticed a few problems with the new web configurator and the icon button widget when using custom icons:

- Custom icons are shown in a circle shape instead of a rectangle with rounded edges like in the icons menu when uploading them
- If you upload larger images the scaled down/cropped icons have aliasing artefacts (see attachment, original file is in UHD resolution)
- If you change the size of the button this results in glitches where the icon gets bigger than the actual button grid cell (see attached video)

### How to Reproduce

1. Create an activity
2. Go to user interface
3. Add a icon button widget and choose a custom icon
4. Resize the size of the button

### Expected behavior

- Custom Icons should look the same in the user interface editor as in the custom icons window after uploading them
- When uploading images larger than the preferred icon size and use the crop function in the upload process images should be scaled down without aliasing artifacts
- When changing the size of an icon widget with a custom icon the size of the icon should be maintained and not extend the edges of the widget

### System version

2.2.1

### What part of the system affected by the problem?

Web Configurator

### Additional context

Aliasing scaling artifacts:
<img width="436" alt="Image" src="https://github.com/user-attachments/assets/4c51729a-43bc-4b77-bcc5-01dca3a82a95" />

Widget scaling glitches:

https://private-user-images.githubusercontent.com/14995555/394955815-e743266e-06af-4d93-9e39-fdb60003ef54.mov?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQxOTYwNTMsIm5iZiI6MTczNDE5NTc1MywicGF0aCI6Ii8xNDk5NTU1NS8zOTQ5NTU4MTUtZTc0MzI2NmUtMDZhZi00ZDkzLTllMzktZmRiNjAwMDNlZjU0Lm1vdj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE0VDE3MDIzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRmMGMwODIzM2NlYTY2MTQ1NDU5MmVmNGZiMzI5MTQ0MTcxMzA3YzFiNDczOThkZDk0ODlmYTk2ZDQ2NTBiNTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bjBSFU2BKm6G4WBF8t8boSQGtJP8MU3UwBr5H1yuxDE

**Comment by kennymc-c** [comment]
According to the 2.2.2 changelog all issues should have been fixed.
But entity and integration icons still have a circle shape while it looks ok in the user interface editor.
Also there are still aliasing artifacts.

Another thing I noticed is that uploading the exact same image with the new configurator results in a softer image and slightly scaled up version (ignore the background not being transparent as already reported in #427, maybe this has something to do with it?)

<img width="220" alt="Image" src="https://github.com/user-attachments/assets/e3dc9f94-b785-4f4d-af36-96f7b8469686" />

**Comment by kennymc-c** [comment]
2.2.3 has fixed the slightly upscaled icon and transparency issues.
But the entity and integration icon shapes are still round and aliasing artifacts can occur.

**Comment by zehnm** [comment]
Fixed in beta release v2.2.3

**Comment by kennymc-c** [comment]
@zehnm No, not all issues have been fixed as mentioned in my previous post

**Comment by martonborzak** [comment]
Icon shape varies as in some UI elements it's masked to a circle for differentiation. This is not a bug.

**Comment by zehnm** [comment]
> If you upload larger images the scaled down/cropped icons have aliasing artefacts (see attachment, original file is in UHD resolution)

The web-configurator provides a resizing tool. This is not a graphics editor and we will not implement different image scaling algorithms. Please use your favorite editor to get pixel perfect results.
Images in the correct size and format will no longer be touched and directly uploaded with the fix in v2.2.3
